### PR TITLE
[UI v2] upgrade: To React 19 🚀🚀🚀

### PR DESCRIPTION
**Pull request by devinvillarosa**
1. Upgrades to react 19 and other dependent packages needed for an upgrade
2. Upgrade `react-day-picker` and [follows migration guide](https://daypicker.dev/upgrading)

Follow: https://react.dev/blog/2024/04/25/react-19-upgrade-guide
`npm install --save-exact react@^19.0.0 react-dom@^19.0.0`
`npm install --save-exact @types/react@^19.0.0 @types/react-dom@^19.0.0`

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] If this pull request adds new functionality, it includes unit tests that cover the changes
- [x] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [x] If this pull request adds functions or classes, it includes helpful docstrings.

Part of https://github.com/PrefectHQ/prefect/issues/15512

### CODE REVIEW

File: ui-v2/package.json [hunk 1]
```diff
@@ -48,11 +48,11 @@
 		"humanize-duration": "^3.32.1",
 		"lucide-react": "^0.447.0",
 		"openapi-fetch": "^0.12.2",
-		"react": "^18.3.1",
-		"react-day-picker": "^8.10.1",
-		"react-dom": "^18.3.1",
+		"react": "19.0.0",
+		"react-day-picker": "^9.4.4",
```
Comment (devinvillarosa) [inline ui-v2/package.json:52]: seems like we don't use this library in the codebase atm. But need to check if this breaks anythiong from v8 to v9

File: ui-v2/src/components/ui/calendar.tsx
Comment (devinvillarosa) [inline ui-v2/src/components/ui/calendar.tsx:1]: Follows this guide: https://daypicker.dev/upgrading

Comment (devinvillarosa) [reply ui-v2/src/components/ui/calendar.tsx:1]: Still need to update the styling a bit.

This should be done in its own PR

**Review by desertaxle — APPROVED** [review]
Sweet!

**Pull request merged**